conversion/conversion.go

```diff
@@ -327,12 +323,22 @@ func dataFromDatabase(ctx context.Context, sourceProfile profiles.SourceProfile,
 
 			// Fetch and store the GCS bucket associated with the datastream
 			dsClient := getDatastreamClient(ctx)
-			gcsBucket, _, fetchGcsErr := streaming.FetchTargetBucketAndPath(ctx, dsClient, targetProfile.Conn.Sp.Project, streamingCfg.DatastreamCfg.DestinationConnectionConfig)
+			gcsBucket, gcsDestPrefix, fetchGcsErr := streaming.FetchTargetBucketAndPath(ctx, dsClient, targetProfile.Conn.Sp.Project, streamingCfg.DatastreamCfg.DestinationConnectionConfig)
 			if fetchGcsErr != nil {
 				logger.Log.Info("Could not fetch GCS Bucket, hence Monitoring Dashboard will not contain Metrics for the gcs bucket\n")
 				logger.Log.Debug("Error", zap.Error(fetchGcsErr))
 			}
 
+			// Try to apply lifecycle rule to Datastream destination bucket.
+			gcsConfig := streamingCfg.GcsCfg
+			if gcsConfig.TtlInDaysSet {
+				err = utils.EnableBucketLifecycleDeleteRule(ctx, gcsBucket, []string{gcsDestPrefix}, gcsConfig.TtlInDays)
```

Comment:
`FetchTargetBucketAndPath` is defined in `streaming` package, while the bucket lifecycle method is defined in `util`. Please refactor this for a more coherent structure.

Reply:
moved FetchTargetBucketAndPath to utils. Ideally we would want a datastream accessor and dataflow accessor (like gcs) but that will be a bigger task.